### PR TITLE
Private Events assotiations

### PR DESCRIPTION
**Pull request by Laguna1**
We built the Private Events application similar to Eventbrite.
This project has no styling because it was not a requirement. All the styles will be added later.
In this app user can:
- sign up, sign in, edit account, and delete the account with the help of the 'devise' gem
- only registered user can to create many events, attend many other`s events created by other users
- each registered user can check his upcoming and previous events
- User’s Show page displays a list of events they are attending
We wrote tests using the 'RSpec '  gem.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### **This PR has been marked as invalid.**

### STATUS: INVALID ❌

REASON: 
- Please enable GitHub actions on your repo and ensure your codebase is properly linted.

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -10,5 +10,37 @@
...
+  <% end %>
+<br/>
+<h3> PAST ATTENDED EVENTS</h3>
+<% current_user.attended_events.each do |event| %>
+  <% if event.date < Date.today %>
+  <ul>
+  <li><%= event.name %> </li>
+  <li><%= event.date %> </li>
+  </ul>
+  <% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: Kindly move logic into views helpers.

Comment (maelfosso) [reply app/views/users/show.html.erb]: You can directly use 
```rb
current_user.attended_events.past_events
```

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -10,5 +10,37 @@
...
+<br/>
+<h3> UPCOMING ATTENDING EVENTS </h3>
+<% current_user.attended_events.each do |event| %>
+  <% if event.date > Date.today %>
+  <ul>
+  <li><%= event.name %> </li>
+  <li><%= event.date %> </li>
+  </ul>
+  <% end %>
+  <% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: Kindly move logic into views helpers
Also, You can directly use 
```rb
current_user.attended_events.future_events
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the job done so far.
However, there are some issues to solve

1. After being logged in, I still have this in the home page
Screenshot at 2020-09-21 14:13:38
You can see the **Logout** button at top at right indicating that I am logged. 
Kindly please change the home page after being logged in

2. Kindly add indexes in **event_attendees** table

3. Kindly move all the logic into views helpers

4. Also add unit and integrations as indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) and [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png)

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, congratulations on addressing all the comments outlined by the previous TSE's :fireworks: 
The project works as expected, test runs perfectly, fixed TSE comments :smile: Congratulations once again.

Go ahead and merge into development, good luck with your next projects :slightly_smiling_face: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**